pycrefine.py

```
@@ -633,16 +875,34 @@ def _close_blocks(self, offset: int):
             offset (int): Bytecode offset used to determine which blocks have ended.
         """
 
-
         while self.blocks and offset >= self.blocks[-1][0]:
             block_end, block_type = self.blocks.pop()
+            
+            has_else = False
+            else_end = None
+            if block_type == "for":
+                for ins in self.instructions:
+                    if ins.offset >= block_end: break
```

Ruff (0.15.9)

[error] Multiple statements on one line (colon)

(E701)